clients/src/main/java/org/apache/kafka/common/requests/FetchRequest.java

```diff
@@ -36,153 +37,162 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
+import java.util.Optional;
 
+import static org.apache.kafka.common.protocol.CommonFields.CURRENT_LEADER_EPOCH;
 import static org.apache.kafka.common.protocol.CommonFields.PARTITION_ID;
 import static org.apache.kafka.common.protocol.CommonFields.TOPIC_NAME;
-import static org.apache.kafka.common.protocol.types.Type.INT32;
-import static org.apache.kafka.common.protocol.types.Type.INT64;
-import static org.apache.kafka.common.protocol.types.Type.INT8;
 import static org.apache.kafka.common.requests.FetchMetadata.FINAL_EPOCH;
 import static org.apache.kafka.common.requests.FetchMetadata.INVALID_SESSION_ID;
 
 public class FetchRequest extends AbstractRequest {
     public static final int CONSUMER_REPLICA_ID = -1;
-    private static final String REPLICA_ID_KEY_NAME = "replica_id";
-    private static final String MAX_WAIT_KEY_NAME = "max_wait_time";
-    private static final String MIN_BYTES_KEY_NAME = "min_bytes";
-    private static final String ISOLATION_LEVEL_KEY_NAME = "isolation_level";
+
     private static final String TOPICS_KEY_NAME = "topics";
     private static final String FORGOTTEN_TOPICS_DATA = "forgotten_topics_data";
-
-    // request and partition level name
-    private static final String MAX_BYTES_KEY_NAME = "max_bytes";
-
-    // topic level field names
     private static final String PARTITIONS_KEY_NAME = "partitions";
 
-    // partition level field names
-    private static final String FETCH_OFFSET_KEY_NAME = "fetch_offset";
-    private static final String LOG_START_OFFSET_KEY_NAME = "log_start_offset";
+    private static final Field.Int32 REPLICA_ID = new Field.Int32("replica_id",
+            "Broker id of the follower. For normal consumers, use -1.");
+    private static final Field.Int64 FETCH_OFFSET = new Field.Int64("fetch_offset", "Message offset.");
+    private static final Field.Int32 PARTITION_MAX_BYTES = new Field.Int32("partition_max_bytes",
```

Comment:
Hmm.. the names are only used internally. Can you clarify how it would impact client developers to change this name?

Reply:
Not sure if it is completely internal. It seems that field name is part of the request specification and request specification is public API. My understanding is that we need to have a KIP for such public API change.

More specifically, custom client library or tool (e.g. https://github.com/linkedin/Burrow) needs to specify the serialization/deserialization of kafka request/response. It seems that developers of these libraries may use kafka source code or documentation as source of truth of the request/response specification. I am a bit concerned that these developers will be confused if we change the name of old schema. Does this make sense?

Reply:
We have not previously considered these field names to be part of the public API, but it's a reasonable question whether we should. Just to be clear, there is no direct dependence on the names in tools like burrow, but your concern is that changing the names may cause confusion? I can appreciate that. This particular name change is probably not strictly needed, but also the potential for confusion either way is probably minimal (it emphasizes but does not change the usage). In the case below, I renamed `epoch` to `session_epoch` because the fetch request now includes the leader epoch as well. I think the potential for confusion is greater if we leave it as is.

Reply:
Yeah I totally agree `partition_max_bytes` is a better name than `max_bytes`. I am wondering if it is better to change the field name for FetchRequest V9 only and keep the name unchanged for FetchRequest V8 and earlier. The benefit of this approach is that developers (both Kafka developer and other library developer) can keep track of the evaluation of the schema (both its structure and field name change). Are you concerned that this would cause confusion if the field name is different for different versions of the schema in `FetchRequest.java`?

I was thinking that field name is part of API because field name is used by developer to identify the request/response schema even though field name is not used by Kafka server code to identify the schema. I may be overly worried. It will be great if we can have input from one more developer to comment on this. What do you think?